proposals/lip-0045.md

```diff
@@ -1575,6 +1446,47 @@ genesisInteroperabilityStoreSchema = {
+* For every `chainInfo` in `chainInfos` with `chainInfo.terminatedState.initialized == False`, the field `chainInfo.terminatedOutbox` must be empty, `chainInfo.terminatedState.stateRoot` must be equal to `EMPTY_HASH` and `chainInfo.terminatedState.mainchainStateRoot` must not be equal to `EMPTY_HASH`. Conversely, if `chainInfo.terminatedState.initialized == True`, then `chainInfo.terminatedState.stateRoot` must not be equal to `EMPTY_HASH` and `chainInfo.terminatedState.mainchainStateRoot` must be equal to `EMPTY_HASH`. todo: must the terminated outbox be non-empty in this case?
+* For each entry `chainInfo` in `chainInfos`, let `chainData = chainInfo.chainData`, then check:
+    * all `chainData.name` are unique.
+    * all `chainData.name` have character set `a-z0-9!@$&_.`.
+    * property `chainData.status` is in set `{CHAIN_STATUS_REGISTERED, CHAIN_STATUS_ACTIVE, CHAIN_STATUS_TERMINATED}`.
+* For each entry `chainInfo` in `chainInfos`, let `activeValidators = chainInfo.chainValidators.activeValidators` and let `certificateThreshold = chainInfo.chainValidators.certificateThreshold`, then check:
+    * `activeValidators` must have at least 1 element and at most `MAX_NUM_VALIDATORS` elements.
+    * `activeValidators` must be ordered lexicographically by `blsKey` property.
+    * All `blsKey` properties must be pairwise distinct.  
+    * Let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`.
+    * Check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division.
+    * Check that the corresponding `validatorsHash` stored in `chainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
+* On a sidechain, if `chainInfos` has an entry for another sidechain (`chainID != getMainchainID()`), then there must also exist an entry for the mainchain (`chainID == getMainchainID`) and `interoperabilityAsset.ownChainInfo` must be non-empty.
+* For each entry `chainInfo` in `chainInfos`, check that `chainInfo.chainData.lastCertificate.timestamp < g.header.timestamp`.
+* For each entry `chainInfo` in `chainInfos`, check that `chainInfo.chainId[0] == interoperabilityAsset.ownChainInfo.chainID[0]` and `chainInfo.chainId != interoperabilityAsset.ownChainInfo.chainID`.
+* On the mainchain, check that `interoperabilityAsset.ownChainInfo.name == CHAIN_NAME_MAINCHAIN` and `interoperabilityAsset.ownChainInfo.chainID == getMainchainID()`.
+* For each entry `chainInfo` in `chainInfos`, either `chainInfo.channelData.messageFeeTokenID == Token.getTokenIDLSK()` (corresponding to the LSK token), or `Token.getChainID(chainInfo.channelData.messageFeeTokenID)` must be equal to `chainInfo.chainID` or `interoperabilityAsset.ownChainInfo.chainID` (the message fee token must be a native token of one of the chains).
+
+##### Genesis State Processing
```

Comment:
Shouldn't this logic be run in the genesis state finalization? I am not sure, as maybe it would be better to completely initialize the internal state already here.